Updated March 2022 for glib 2.72.0 by @pkgw
Updated November 2021 for glib 2.70.1 by @pkgw

From 41a15be2e45ccfdfac236cc043a1e2c7d5f73bfa Mon Sep 17 00:00:00 2001
From: Marius van Niekerk <marius.v.niekerk@gmail.com>
Date: Sat, 2 Mar 2019 17:34:43 -0500
Subject: [PATCH] Attempt to mask out qemu failing tests

diff --git a/glib/tests/meson.build b/glib/tests/meson.build
index a0c64af..282540a 100644
--- a/glib/tests/meson.build
+++ b/glib/tests/meson.build
@@ -35,7 +35,7 @@ glib_tests = {
   },
   'guuid' : {},
   'gvariant' : {
-    'suite' : ['slow'],
+    'suite' : ['slow', 'flaky'],
   },
   'gwakeup' : {
     'source' : ['gwakeuptest.c', '../gwakeup.c'],
@@ -102,7 +102,7 @@ glib_tests = {
   'spawn-multithreaded' : {},
   'spawn-path-search' : {},
   'spawn-singlethread' : {},
-  'strfuncs' : {},
+  'strfuncs' : {'suite': ['flaky']},
   'string' : {},
   'strvbuilder' : {},
   'testing' : {},
@@ -128,12 +128,12 @@ glib_tests = {
     'suite' : ['slow'],
   },
   '642026' : {
-    'suite' : ['slow'],
+    'suite' : ['slow', 'flaky'],
   },
   '642026-ec' : {
     'source' : '642026.c',
     'c_args' : ['-DG_ERRORCHECK_MUTEXES'],
-    'suite' : ['slow'],
+    'suite' : ['slow', 'flaky'],
   },
 }
 
diff --git a/gio/tests/meson.build b/gio/tests/meson.build
index 055cade..858600c 100644
--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -101,6 +101,7 @@ gio_tests = {
   'socket' : {
     # FIXME: https://gitlab.gnome.org/GNOME/glib/-/issues/1392
     'should_fail' : host_system == 'darwin',
+    'suite': ['flaky']
   },
   'socket-listener' : {},
   'socket-service' : {},
diff --git a/gobject/tests/meson.build b/gobject/tests/meson.build
index 8d568f0..43b916f 100644
--- a/gobject/tests/meson.build
+++ b/gobject/tests/meson.build
@@ -124,7 +124,7 @@ foreach test_name : python_tests
     python,
     args: ['-B', files(test_name)],
     env: test_env,
-    suite: ['gobject', 'no-valgrind'],
+    suite: ['gobject', 'no-valgrind', 'flaky'],
   )
 
   if installed_tests_enabled
